MuonAnalysis/MuonAssociators/src/PropagateToMuonSetup.cc

```diff
@@ -0,0 +1,81 @@
+      useMB2_(iConfig.existsAs<bool>("useStation2") ? iConfig.getParameter<bool>("useStation2") : true),
+      fallbackToME1_(iConfig.existsAs<bool>("fallbackToME1") ? iConfig.getParameter<bool>("fallbackToME1") : false),
+      whichTrack_(None),
+      whichState_(AtVertex),
+      cosmicPropagation_(iConfig.existsAs<bool>("cosmicPropagationHypothesis")
+                             ? iConfig.getParameter<bool>("cosmicPropagationHypothesis")
+                             : false),
+      useMB2InOverlap_(iConfig.existsAs<bool>("useMB2InOverlap") ? iConfig.getParameter<bool>("useMB2InOverlap")
+                                                                 : false),
+      magfieldToken_(iC.esConsumes()),
+      propagatorToken_(iC.esConsumes(iConfig.getParameter<edm::ESInputTag>("propagatorAlong"))),
+      propagatorAnyToken_(iC.esConsumes(iConfig.getParameter<edm::ESInputTag>("propagatorAny"))),
+      propagatorOppositeToken_(iC.esConsumes(iConfig.getParameter<edm::ESInputTag>("propagatorOpposite"))),
+      muonGeometryToken_(iC.esConsumes()) {
+  std::string whichTrack = iConfig.getParameter<std::string>("useTrack");
+  if (whichTrack == "none") {
+    whichTrack_ = None;
+  } else if (whichTrack == "tracker") {
+    whichTrack_ = TrackerTk;
+  } else if (whichTrack == "muon") {
+    whichTrack_ = MuonTk;
+  } else if (whichTrack == "global") {
+    whichTrack_ = GlobalTk;
+  } else
+    throw cms::Exception("Configuration") << "Parameter 'useTrack' must be 'none', 'tracker', 'muon', 'global'\n";
+  if (whichTrack_ != None) {
+    std::string whichState = iConfig.getParameter<std::string>("useState");
+    if (whichState == "atVertex") {
+      whichState_ = AtVertex;
+    } else if (whichState == "innermost") {
+      whichState_ = Innermost;
+    } else if (whichState == "outermost") {
+      whichState_ = Outermost;
+    } else
+      throw cms::Exception("Configuration") << "Parameter 'useState' must be 'atVertex', 'innermost', "
+                                               "'outermost'\n";
+  }
+  if (cosmicPropagation_ && (whichTrack_ == None || whichState_ == AtVertex)) {
+    throw cms::Exception("Configuration") << "When using 'cosmicPropagationHypothesis' useTrack must not be "
+                                             "'none', and the state must not be 'atVertex'\n";
+  }
+}
+
+PropagateToMuon PropagateToMuonSetup::init(const edm::EventSetup &iSetup) const {
+  auto const magfield = iSetup.getHandle(magfieldToken_);
+  auto const propagator = iSetup.getHandle(propagatorToken_);
+  auto const propagatorOpposite = iSetup.getHandle(propagatorOppositeToken_);
+  auto const propagatorAny = iSetup.getHandle(propagatorAnyToken_);
+  auto const muonGeometry = iSetup.getHandle(muonGeometryToken_);
+
+  return PropagateToMuon(magfield,
+                         propagator,
+                         propagatorOpposite,
+                         propagatorAny,
```

Comment:
```suggestion
                         propagatorAny,
                         propagatorOpposite,
```
In the ctor of `PropagateToMuon`, `Any` comes `Opposite`, so looks like they should be swapped here.

Reply:
True, thanks for catching that. Done.